### PR TITLE
RSS-ECOMM-2_09 and RSS-ECOMM-2_10

### PR DESCRIPTION
**Pull request by FedAliaks**
# Description

- (25 points) Implement client-side validation for all required fields in the registration form, such as email, password, first name, last name, date of birth, and address fields (e.g., street, city, postal code, and country) for proper use with CommerceTools. [RSS-ECOMM-2_09](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint2/RSS-ECOMM-2_09.md)
- (20 points) Display clear error messages indicating any validation issues, such as an improperly formatted email or a weak password. [RSS-ECOMM-2_10](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint2/RSS-ECOMM-2_10.md)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings (I have one console.log for future features)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: rs-commerce/src/views/registration/components/RegistrationForm/components/AccautnRegistration/AccountRegistration.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        }}
+      />
+      <InputRegistration
+        input={{
+          htmlFor: 'dateOfBirth',
+          title: 'Date of birth:',
+          type: 'date',
+          placeholder: '',
+        }}
+      />
```
Comment (DaryaAniskevich) [inline rs-commerce/src/views/registration/components/RegistrationForm/components/AccautnRegistration/AccountRegistration.tsx]: You can create array with input fields objects and use `map` to create Inputs elements from thia array

File: rs-commerce/src/views/registration/components/RegistrationForm/components/AddressRegistration/AddressRegistration.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+function AddressRegistration() {
+  return (
+    <div className={classes['form__column']}>
+      <div className={classes['billing__container']}>
+        <div className={classes['registration__subtitle-block']}>
+          <h2 className={classes['registration__column-title']}>Address</h2>
+          <label className={classes['input__label']} htmlFor="setSameAddress">
+            <input className={classes['input_checkbox']} type="checkbox" id="setSameAddress" />
+            <p className={classes['checkbox__content']}>One address for billing and shipping.</p>
+          </label>
```
Comment (DaryaAniskevich) [inline rs-commerce/src/views/registration/components/RegistrationForm/components/AddressRegistration/AddressRegistration.tsx]: You can create separate reusable component for checkbox

File: rs-commerce/src/views/registration/components/RegistrationForm/components/AddressRegistration/components/AddressComponents/AddressComponents.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+        <InputRegistration
+          input={{
+            htmlFor: `${typeComponent}Street`,
+            title: 'Street:',
+            type: 'text',
+            placeholder: 'your street',
+            smallSize: true,
+          }}
+        />
```
Comment (DaryaAniskevich) [inline rs-commerce/src/views/registration/components/RegistrationForm/components/AddressRegistration/components/AddressComponents/AddressComponents.tsx]: You can use array and map it here too

File: rs-commerce/src/views/registration/components/RegistrationForm/components/AddressRegistration/components/AddressComponents/components/AddresTitleComponent/AddresTitleComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    <div className={classes['address__title-block']}>
+      <h3 className={classes['address__subtitle']}>{`${typeComponentUpper} address`}</h3>
+      <label className={classes['input__label']} htmlFor={`set${typeComponentUpper}DefaultAddress`}>
+        <input
+          className={classes['input_checkbox']}
+          type="checkbox"
+          id={`set${typeComponentUpper}DefaultAddress`}
+        />
+        <p className={classes['checkbox__content']}>Set as a default address</p>
+      </label>
```
Comment (DaryaAniskevich) [inline rs-commerce/src/views/registration/components/RegistrationForm/components/AddressRegistration/components/AddressComponents/components/AddresTitleComponent/AddresTitleComponent.tsx]: You can use separate Checkbox component here the same like in `AddressRegistration` component. Difference will be in props

File: rs-commerce/src/views/registration/components/RegistrationForm/components/ButtonRegistration/ButtonRegistration.tsx
Comment (DaryaAniskevich) [inline rs-commerce/src/views/registration/components/RegistrationForm/components/ButtonRegistration/ButtonRegistration.tsx:1]: I suppose you can use Submit Button component which Sergey has created for Login page and pass your click handler there

File: rs-commerce/src/views/registration/components/RegistrationForm/components/InputRegistration/CountryInput.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        <input
+          id={`${typeComponent}Country`}
+          className={`${classes['registration__input']} ${classes['registration__input_small']}`}
+          onChange={(e) => getValue(e)}
+          list="country"
+        />
+        <datalist id="country">
+          <option>USA</option>
+          <option>Belarus</option>
+        </datalist>
```
Comment (DaryaAniskevich) [inline rs-commerce/src/views/registration/components/RegistrationForm/components/InputRegistration/CountryInput.tsx]: Use array of countries to map options

File: rs-commerce/src/views/registration/components/RegistrationForm/components/AccautnRegistration/AccountRegistration.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      placeholder: 'your last name',
+    }}
+  />,
+  <InputRegistration
+    input={{
+      htmlFor: 'dateOfBirth',
+      title: 'Date of birth:',
+      type: 'date',
+      placeholder: '',
+    }}
```
Comment (DaryaAniskevich) [inline rs-commerce/src/views/registration/components/RegistrationForm/components/AccautnRegistration/AccountRegistration.tsx]: we don't need components array. create array like 
`[{   htmlFor: 'lastName',
      title: 'Last name:',
      type: 'text',
      placeholder: 'your last name',
    }, {htmlFor: 'dateOfBirth',
      title: 'Date of birth:',
      type: 'date',
      placeholder: '',}]`

and map it `{inputFieldsArray.map((item) => <InputRegistration
    input={{
      htmlFor: item.htmlFor,
      title: item.title,
      type: item.type,
      placeholder: item.placeholder,
    }}
  />)},`

File: rs-commerce/src/views/registration/components/RegistrationForm/components/AddressRegistration/components/AddressComponents/AddressComponents.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  <InputRegistration
+    input={{
+      htmlFor: `${typeComponent}Street`,
+      title: 'Street:',
+      type: 'text',
+      placeholder: 'your street',
+      smallSize: true,
+    }}
+  />,
+];
```
Comment (DaryaAniskevich) [inline rs-commerce/src/views/registration/components/RegistrationForm/components/AddressRegistration/components/AddressComponents/AddressComponents.tsx]: the same like in `AccountRegistration`

**Review by sergioivanov008 — APPROVED** [review]
Approve

**Review by DaryaAniskevich — APPROVED** [review]

**Pull request merged**